articles/build-a-restful-api-with-adonisjs/index.md

```diff
@@ -0,0 +1,419 @@
+### Build A RESTful API With AdonisJS
+
+### Outline
+Introduction
+Setting up AdonisJS
+Creating Databases
+Setting up Authorization and Authentication
+Creating Models
+Creating Controllers
+Creating Endpoint Routes
+Testing the Forum API
+Conclusion
+
+### Introduction
+In today’s ever-changing and rapid world of frameworks including frontend and backend frameworks, learning and building RESTful APIs to bridge the gap between clients and data is a demanding skill for any software engineer.
+
+"A RESTful API is an application program interface (API) that uses HTTP requests exchange data between parties using endpoints (routes) with either GET, PUT, POST and DELETE as the HTTP requests." from [TowardDataScience](https://towardsdatascience.com)
+
+In this tutorial, we will be showing you how to build a RESTful API with AdonisJS 5. we will be building an API for a forum, we are going to cover a lot such as authentication, authorization, etc. we will also cover best practices in structuring and building out your HTTPS endpoint and also industry standard responses.
+
+AdonisJS is the Laravel of JavaScript and a good grasp of it is a high-demand skill, therefore learning a build your first REST API with it is a game-changer. You can read through the [Ultimate Guide to AdonisJS 5](https://masteringbackend.com/posts/adonisjs-tutorial-the-ultimate-guide) to learn more.
+
+### Setting up AdonisJS 5
+If you’ve been working with JavaScript frameworks before now, you should already have Node.js in your computer, if not, you can install the latest version by going through the steps [here](https://nodejs.org/en/).
```

Comment:
`If you’ve been working with JavaScript frameworks before,` sounds better. Remove now.

Reply:
done